### DISCUSSION TITLE
Installing borg as secondary package manager : clone epkgs issue and git staging issue.

### DISCUSSION THREAD

**Discussion by rileyrg** [emacscollective/borg#107] [Support] [answered]
reference : https://emacsmirror.net/manual/borg/Use-as-secondary-package-manager.html#Use-as-secondary-package-manager

1) observation (possibly my setup specific) : When I tried to install/assimilate my first package, in this case json-navigator, borg prompted me to git clone epkgs. Unfortunately the buffer it prompts in in isn't interactive. I killed the buffer and cloned ekpgs directly from the command line.

2) Git : so I have assimilated my first package as outlined above, but git is confusing me a little. Since I use an exclusionary .gitignore for my existing emacs config, I enabled borg and borg/* - it insisted on that in order to assimiliate.  If I dont, I see

> The following paths are ignored by one of your .gitignore files:
> borg/json-navigator
> hint: Use -f if you really want to add them.
> hint: Turn this message off by running
> hint: "git config advice.addIgnoredFile false"
>
 
So in my .gitignore

> !borg
> !borg/*

I'm assuming here that we do indeed want to add them. But I'm not 100% sure as I don't use git submodules until now.

After this I can assimilate fine:

> (09:14:27) Building json-navigator
> 
> Initializing drones...done (1 drones in 0.004s)
> Package cl is deprecated
>  Creating /home/rgr/Dropbox/homefiles/directories/development/projects/emacs/emacs-config/borg/json-navigator/json-navigator-autoloads.el...
> Compiling /home/rgr/Dropbox/homefiles/directories/development/projects/emacs/emacs-config/borg/json-navigator/json-navigator.el...
> Done (Total of 1 file compiled)
> 
> Process emacs ... --eval (borg-build "json-navigator") finished
> 

My magit status is now (excuse the screenshot - I'm unable to copy selected region as text from magit status)

![screenshot-latest](https://user-images.githubusercontent.com/362891/120282300-3ee78d00-c2ba-11eb-95e3-2503b9009797.png)

If I try to stage unstaged changes The borg module won't stage. And expanding, I see this,

![screenshot-latest](https://user-images.githubusercontent.com/362891/120282680-a998c880-c2ba-11eb-9218-5937dc936ca9.png)

I'm guessing this is a trivial git issue I don't understand regarding borg directory's .gitignore status. Thanks for the help.

**Comment by tarsius** [comment, marked as answer]
1. The "Clone https://github.com/emacsmirror/epkgs.git to ~/.emacs.d/epkgs/? " prompt? What do you mean by "isn't interactive"? Anyway, sounds like a fluke and you got it going.

2. Yeah, that's expected and your approach to addressing seems appropriate.  But there's a second step: (goto 3)

3. While the cursor is on "modified    borg/json-navigator" press enter to go to the submodule. You will probably "learn" that there is a new untracked file "json-navigator.elc", and some others.  Setup a global gitignore rule for that as described at the end of https://emacsmirror.net/manual/borg/Bootstrapping-using-a-seed.html.

**Reply by rileyrg** [reply]
> The "Clone https://github.com/emacsmirror/epkgs.git to ~/.emacs.d/epkgs/? " prompt? What do you mean by "isn't interactive"? Anyway, sounds like a fluke and you got it going.

It means its displayed in a buffer asking for a "yes or no" but there is no yes/no prompt. Non-interactive. I tried from scratch twice and both times it failed at this point. So I cloned from the command line.

> While the cursor is on "modified borg/json-navigator" press enter to go to the submodule. You will probably "learn" that there is a new untracked file "json-navigator.elc", and some others. Setup a global gitignore rule for that as described at the end of https://emacsmirror.net/manual/borg/Bootstrapping-using-a-seed.html.

> . For this reason it is useful to ignore these outputs globally, for example in your ~/.config/git/ignore file:

I had added something similar to .git/ignore/exclude. Did you mean exclude?   I had a quick google and exclude seems to be the file for this functionality. 

Anyway : sods law. I assimilated two larger packages in lsp-mode and magit - and they both worked. It seems that json-navigator is the exception in my test cases and that and only that cause this issue so far. Sigh.

So, just about up and running now. Thank you.

**Reply by tarsius** [reply]
> I had added something similar to .git/ignore/exclude. Did you mean exclude?

No, I mean `$XDG_CONFIG_HOME/git/ignore`. See `gitignore` manpage.

**Reply by rileyrg** [reply]
Aha. OK Good time to move to an XDG_CONFIG based git config. Working. Thank you.
